src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,159 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int quantity = 0;
+    private Node<T> node;
+    private Node<T> temp;
+
     @Override
     public void add(T value) {
+        rewindNodeToHead();
+        if (quantity == 0) {
+            node = new Node(null, value, null, true, true);
+        } else {
+            rewindToLastNode();
+            node.setTail(false);
+            node.setNext(new Node<>(node, value, null, false, true));
+        }
+        quantity += 1;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > quantity || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index");
+        }
+        rewindNodeToHead();
+        if (index == quantity) {
+            add(value);
+        } else {
+            rewindToSelectedNode(index);
+            insertNodeInCurrentPosition(index, value);
+            quantity += 1;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= quantity || index < 0) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        rewindNodeToHead();
+        for (int i = 0; i < quantity; i++) {
+            if (i != index) {
+                node = node.getNext();
+                continue;
+            }
+            temp = node;
+        }
+        return temp.getElement();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index >= quantity || index < 0) {
+            throw new IndexOutOfBoundsException("Incorrect index");
+        }
+        rewindNodeToHead();
+        rewindToSelectedNode(index);
+        T res = node.getElement();
+        node.setElement(value);
```

Comment:
Don't use getters and setters in class Node. You have direct access to the fields of the inner class Node, so using setters like setElement is unnecessary.